www-client/firefox/firefox-115.11.0.ebuild

```diff
@@ -114,10 +114,13 @@ BDEPEND="${PYTHON_DEPS}
 	>=dev-util/cbindgen-0.24.3
 	net-libs/nodejs
 	virtual/pkgconfig
-	!clang? ( >=virtual/rust-1.65 )
+	!clang? (
+		>=virtual/rust-1.65
+		<virtual/rust-1.78
+	)
 	!elibc_glibc? (
 		|| (
-			dev-lang/rust
+			<dev-lang/rust-1.78
 			<dev-lang/rust-bin-1.73
 		)
 	)
```

Comment:
_Hello everyone,_

Could someone clarify why this mix of restrictions might be used for `!elibc_glibc?`?
I mean, why not use `virtual/rust` instead of `dev-lang/rust{,-bin}` like for `!clang?` and why `rust-bin` is limited to `<1.73`? Similar questions are relevant for other ebuilds.

_Thanks!_

Reply:
You can check git commit history but rust-bin is broken (for reasons unknown, someone needs to investigate it) with musl with newer versions for the purposes of building FF.